daemon/web/components/App.js

```diff
@@ -6,29 +6,74 @@ import InertiaClient from '../client';
 export default class App extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      username: "",
+      password: "",
+      loginAlert: ""
+    };
+    this.handleLoginSubmit = this.handleLoginSubmit.bind(this);
+    this.handleUsernameBlur = this.handleUsernameBlur.bind(this);
+    this.handlePasswordBlur = this.handlePasswordBlur.bind(this);
+  }
+
+  async handleLoginSubmit() {
+    const endpoint = 'https://localhost:8081/web/login';
```

Comment:
Note that `client.js` has the [daemon host path](https://github.com/ubclaunchpad/inertia/blob/john/%23137-web-login/daemon/web/client.js#L5) which is [dynamically determined](https://github.com/ubclaunchpad/inertia/blob/john/%23137-web-login/daemon/web/index.js#L8). The client is then [passed to the App](https://github.com/ubclaunchpad/inertia/blob/john/%23137-web-login/daemon/web/index.js#L10).

```js
const daemonAddress = window.location.host;
const client = new InertiaClient(daemonAddress);
ReactDOM.render(<App client={client}/>, document.getElementById('app'));
```

Instead of having this function here, i think we should place all API calls in `client.js` and access it through the `InertiaClient` class - this allows us to avoid having the define endpoints everywhere

Another note: I don't think `localhost` works very well because of the way the Docker routing works, have you tried using `0.0.0.0:8081/web/login`? See #126

Reply:
I modified the code to use the `daemonAddress`.

However, `window.location.host` becomes `http://localhost:7900` when I run it in dev mode. So I've added a `NODE_ENV` environment variable to webpack as a temporary measure for now.

Reply:
I suppose an option would be to have `make web-build` pass along the daemon url as an env variable.

Reply:
I think the way you set it up as is fine (with the node env)